a_test.go

```diff
@@ -183,42 +185,374 @@ func TestA_uses_Logger_dynamic_interface(t *testing.T) {
 	}
 }
 
+func TestA_WithContext(t *testing.T) {
+	t.Parallel()
+	type ctxKeyT struct{}
+	var ctxKey ctxKeyT
+
+	for _, parallel := range []bool{false, true} {
+		parallel := parallel
+		for _, throwPanic := range []bool{false, true} {
+			throwPanic := throwPanic
+			t.Run("Parallel="+strconv.FormatBool(parallel)+" and panic="+strconv.FormatBool(throwPanic), func(t *testing.T) {
+				t.Parallel()
+
+				flow := &goyek.Flow{}
+
+				flow.SetOutput(io.Discard)
+				loggerSpy := &helperLoggerSpy{}
+				flow.SetLogger(loggerSpy)
+
+				depsTask := flow.Define(goyek.Task{
+					Name: "deps",
+					Action: func(a *goyek.A) {
+						a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+						onceCall(t, a.Name())()
+					},
+				})
+
+				childTasks := make(goyek.Deps, 10)
+				childTasksResults := make([]bool, len(childTasks))
+
+				task := flow.Define(goyek.Task{
+					Name: "task",
+					Action: func(a *goyek.A) {
+						a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+						onceCall(t, a.Name())()
+
+						k, ok := a.Context().Value(ctxKey).(int)
+						if !ok {
+							return
+						}
+
+						childTasksResults[k] = true
+					},
+					Deps: goyek.Deps{depsTask},
+				})
+
+				for i := 0; i < len(childTasks); i++ {
+					i := i
+					childTasks[i] = flow.Define(goyek.Task{
+						Name: "child task " + strconv.Itoa(i),
+						Action: func(a *goyek.A) {
+							a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+							onceCall(t, a.Name())()
+
+							if throwPanic {
+								a.Cleanup(func() {
+									panic("some panic")
+								})
+							}
```

Comment:
Conditional logic like this makes tests very hard to maintain. I think there are two ways to solve it:
- duplicating the logic (adding a new test function e.g. `TestA_WithContextPanic`)
- adding some `func()` test case parameters to have common code for all `TestA_WithContext`, `TestA_WithContextFatal`, `TestA_WithContextSkipped`

I think it would be good to try to add some test case parameters as function hooks.

Reply:
They are similar in many ways, but also similar in places... I try to unite them with helpers.

Reply:
Consider time-boxing the effort. If uniting would be not straightforward, then I prefer more readable duplicated code. We can look into refactoring the duplication in a separate PR.